src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int DOUBLE_SIZE = 2;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_SIZE];
+    }
 
     @Override
     public void put(K key, V value) {
+        resize();
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = getBucketIndex(key);
+        Node<K, V> node = buckets[index];
+
+        if (node == null) {
+            buckets[index] = newNode;
+            size++;
 
+        } else {
+            while (node != null) {
+                if ((node.key == null && key == null || node.key != null && node.key.equals(key))) {
```

Comment:
let's extract this key comparison logic to a separate private method
```suggestion
                if ((node.key == key || node.key != null && node.key.equals(key))) {
```

Reply:
ok